frontend/src/metabase/reference/containers/FieldXray.jsx

```diff
@@ -0,0 +1,165 @@
+/* @flow */
+import React, { Component } from 'react'
+import Visualization from 'metabase/visualizations/components/Visualization'
+
+import { connect } from 'react-redux'
+
+import type { Field } from 'metabase/meta/types/Field'
+import { fetchFieldFingerPrint } from 'metabase/reference/reference'
+
+type Props = {
+    fetchXrayData: () => void,
+    field: Field,
+    xray: {}
+}
+
+class FieldXray extends Component {
+    props: Props
+
+    componentWillMount() {
+        this.props.fetchXrayData(this.props.field.id)
+    }
+    
+    // componentDidMount () {
+    //     // QUESTION: What am I doing wrong here?
+    //     let fingerprint = await this.props.fetchFieldFingerprint(this.props.fieldId)
+    //     this.setState({ fingerprint })
+    // }
+
+    render () {
+        return (
+            <div style={{
+                maxWidth: 550,
+                marginLeft: 'calc(48px + 1rem)',
+            }}>
+                <h3>XRAY</h3>
+                <div className="Grid Grid--1of2 Grid--gutters mt1">
+                    <div className="Grid-cell">
+                        <div className="bordered rounded shadowed py2">
+                            <Visualization
+                                series={[this.props.xray.min]}
+                            />
+                        </div>
+                    </div>
+                    <div className="Grid-cell">
+                        <div className="bordered rounded shadowed py2">
+                            <Visualization
+                                series={[this.props.xray.max]}
+                            />
+                        </div>
+                    </div>
+                </div>
+                <Visualization
+                    series={[this.props.xray.histogram]}
+                />
+                <div>{[this.props.fingerprint]}</div>
+            </div>
+        )
+    }
+}
+
+const mapStateToProps = state => ({
+    xray: {
+        min: {
+            card: {
+                name: 'Min',
+                display: 'scalar'
+            },
+            data: {
+                cols: [{
```

Comment:
Most of the columns metadata can be left out. I think `type` and `name` are the important parts.